src/app/capture/page.tsx

```diff
@@ -6,9 +6,39 @@ import styles from './page.module.css';
 import DeerCamera from '@/components/deer-camera/DeerCamera';
 import Loader from '@/components/loader/Loader';
 import BottomNav from '@/components/bottom-nav/BottomNav';
+import { cooldownTime } from '@/utils/constants';
+import { useMutation } from '@tanstack/react-query';
+import { createDeerSighting } from '@/hooks/apis/useDeerSighting';
+import { useLocationContext } from '@/context/LocationContext';
 
 export default function Capture() {
   const [loading, setLoading] = useState(true);
+  const { userLocation } = useLocationContext();
+
+  const useCreateDeerSighting = useMutation({
+    mutationFn: createDeerSighting,
+    onSuccess: (res) => {
+      console.log('Deer Saved to System');
+      console.log(res);
+    },
+    onError: (err) => {
+      console.log(err);
+    },
+  });
+
+  const SaveDeer = useCallback(() => {
+    if (userLocation?.latitude && userLocation?.longitude) {
+      const curDate = new Date().toDateString();
+
+      useCreateDeerSighting.mutate({
+        longitude: userLocation.longitude,
+        latitude: userLocation.latitude,
+        timestamp: curDate,
+      });
+    } else {
+      console.log('Location is invalid');
+    }
```

Comment:
Do you think we can create some sort of refitting/retrying callback function on failure? so it will try to refetch the info. If you agree and think that should be a story then def go ahead and add it

Reply:
Thats a good call, with the story where we add a timer it would refetch the location automatically regardless of whether the location is invalid or not, so I did withheld adding that for this story

Reply:
to add, if it does refetch and its invalid within that other story, it wont have the location be null it'll just keep the last good coordinates it has saved instead